internal/storage/storage_handle_test.go

```diff
@@ -5,9 +5,63 @@ import (
 	"testing"
 	"time"
 
+	"github.com/fsouza/fake-gcs-server/fakestorage"
 	"golang.org/x/oauth2"
 )
 
+const correctBucketName string = "will-be-present-in-fake-server"
+const wrongBucketName string = "will-not-be-present-in-fake-server"
+
+func createFakeServer() (fakeServer *fakestorage.Server, err error) {
+	fakeServer, err = fakestorage.NewServerWithOptions(fakestorage.Options{
+		InitialObjects: []fakestorage.Object{
+			{
+				ObjectAttrs: fakestorage.ObjectAttrs{
+					BucketName: correctBucketName,
+				},
+			},
+		},
+		Host: "127.0.0.1",
+		Port: 8081,
+	})
+	return
+}
+
+func TestBucketHandleWhenBucketExists(t *testing.T) {
+	server, err := createFakeServer()
+	if err != nil {
+		t.Fatalf("Server creation failed")
+	}
+	defer server.Stop()
+
+	fakeClient := server.Client()
+	fakeStorageClient := &StorageClient{client: fakeClient}
+	bucketHandle, err := fakeStorageClient.BucketHandle(correctBucketName)
+
+	if err != nil {
```

Comment:
given that we are going to use 3rd party libraries for unit tests, go/gcs-clients-coding-guidelines on testing is not valid. Please use ooglematchers/ooglemock like in other test files to do asserting.

Sorry for back and forth.

Reply:
But here we are using feature specific 3rd party libraries, not like ooglematchers/ooglemock which was specially for testing (feature agnostic).

Reply:
Right. the guideline going forward is " We are okay to use 3rd party libraries for unit testing". And since we already use ooglematchers/ooglemock in gcsfuse, lets use it here also.

Reply:
Sure.